tests/AbstractLoopTest.php

```diff
@@ -398,6 +398,35 @@ function () {
         $this->loop->run();
     }
 
+    public function testSignal()
+    {
+        if (!extension_loaded('pcntl')) {
+            $this->markTestSkipped('Signal test skipped because PCNTL extension isn\'t loaded.');
+        }
+
+        $called = false;
+        $calledShouldNot = true;
+
+        $this->loop->onSignal(SIGUSR2, $func2 = function () use (&$calledShouldNot) {
+            $calledShouldNot = false;
+        });
+
+        $this->loop->onSignal(SIGUSR1, $func1 = function () use (&$func1, &$func2, &$called) {
+            $called = true;
+            $this->loop->removeSignal(SIGUSR1, $func1);
+            $this->loop->removeSignal(SIGUSR2, $func2);
+        });
+
+        $this->loop->futureTick(function () {
+            posix_kill(posix_getpid(), SIGUSR1);
```

Comment:
Given this is a unit test I'm not that worried. Might put an if and fail in the if the function isn't available